server/bleep/src/cache.rs

```diff
@@ -137,38 +260,20 @@ impl<'a> FileCache<'a> {
         Ok(())
     }
 
-    async fn delete_files(&self, tx: &mut sqlx::Transaction<'_, Sqlite>) -> anyhow::Result<()> {
-        let repo_str = self.reporef.to_string();
-        sqlx::query! {
-            "DELETE FROM file_cache \
-                 WHERE repo_ref = ?",
-            repo_str
-        }
-        .execute(&mut *tx)
-        .await?;
-
-        Ok(())
-    }
-
-    async fn delete_chunks(&self, tx: &mut sqlx::Transaction<'_, Sqlite>) -> anyhow::Result<()> {
-        let repo_str = self.reporef.to_string();
-        sqlx::query! {
-            "DELETE FROM chunk_cache \
-                 WHERE repo_ref = ?",
-            repo_str
-        }
-        .execute(&mut *tx)
-        .await?;
-
-        Ok(())
+    /// Process the next chunk from the embedding queue if the batch size is met.
+    pub fn process_embedding_queue(&self) -> anyhow::Result<()> {
+        tokio::task::block_in_place(|| {
+            tokio::runtime::Handle::current()
+                .block_on(async { self.batched_embed_or_flush_queue(false).await })
+        })
     }
 
     /// Commit the embed log, invoking the embedder if batch size is met.
     ///
     /// If `flush == true`, drain the log, send the entire batch to
     /// the embedder, and commit the results, disregarding the internal
     /// batch sizing.
-    pub async fn batched_process_embed_queue(&self, flush: bool) -> anyhow::Result<()> {
+    async fn batched_embed_or_flush_queue(&self, flush: bool) -> anyhow::Result<()> {
```

Comment:
Can we just merge this with the function below?

Reply:
It will make things fairly complicated, and that loop is quite dense as is. This is supposed to be an easy to grasp outline of how we generate embeddings, while the lower layer function is dealing with actually batching & making those embeddings.